scripts/install_third_party_libs.py

```diff
@@ -88,72 +84,382 @@ def make_google_module_importable_by_python(google_module_path: str) -> None:
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE)
+        print('Redis-cli is already installed.')
+    except OSError:
+        # The redis-cli is not installed, run the script to install it.
+        # NOTE: We do the installation here since we need to use make.
+        print('Installing redis-cli...')
+
+        download_and_untar_files(
+            ('https://download.redis.io/releases/redis-%s.tar.gz') %
+            common.REDIS_CLI_VERSION,
+            common.OPPIA_TOOLS_DIR,
+            'redis-%s' % common.REDIS_CLI_VERSION,
+            'redis-cli-%s' % common.REDIS_CLI_VERSION)
+
+        # Temporarily change the working directory to redis-cli-6.0.6 so we can
+        # build the source code.
+        with common.CD(
+            os.path.join(
+                common.OPPIA_TOOLS_DIR,
+                'redis-cli-%s' % common.REDIS_CLI_VERSION)):
+            # Build the scripts necessary to start the redis server.
+            # The make command only builds the C++ files in the src/ folder
+            # without modifying anything outside of the oppia root directory.
+            # It will build the redis-cli and redis-server files so that we can
+            # run the server from inside the oppia folder by executing the
+            # script src/redis-cli and src/redis-server.
+            subprocess.call(['make'])
+
+        # Make the scripts executable.
+        subprocess.call([
+            'chmod', '+x', common.REDIS_SERVER_PATH])
+        subprocess.call([
+            'chmod', '+x', common.REDIS_CLI_PATH])
+
+        print('Redis-cli installed successfully.')
+
+
+def install_elasticsearch_dev_server() -> None:
+    """This installs a local ElasticSearch server to the oppia_tools
+    directory to be used by development servers and backend tests.
+    """
+    try:
+        subprocess.call(
+            ['%s/bin/elasticsearch' % common.ES_PATH, '--version'],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            # Set the minimum heap size to 100 MB and maximum to 500 MB.
+            env={'ES_JAVA_OPTS': '-Xms100m -Xmx500m'}
+        )
+        print('ElasticSearch is already installed.')
+        return
+    except OSError:
+        print('Installing ElasticSearch...')
+
+    if common.is_mac_os() or common.is_linux_os():
+        download_and_untar_files(
+            'https://artifacts.elastic.co/downloads/elasticsearch/' +
+            'elasticsearch-%s-%s-x86_64.tar.gz' % (
+                common.ELASTICSEARCH_VERSION,
+                common.OS_NAME.lower()
+            ),
+            common.OPPIA_TOOLS_DIR,
+            'elasticsearch-%s' % common.ELASTICSEARCH_VERSION,
+            'elasticsearch-%s' % common.ELASTICSEARCH_VERSION
+        )
+
+    else:
+        raise Exception('Unrecognized or unsupported operating system.')
+
+    print('ElasticSearch installed successfully.')
+
+
 def main() -> None:
```

Comment:
Note to reviewers: This is the main function to look at in order to understand the installation flow. All install steps are now included here rather than spread out across several files.